src/components/AnchoredTextLabels.cpp

```diff
@@ -0,0 +1,61 @@
+#include "AnchoredTextLabels.h"
+#include <Components/BsCCamera.h>
+#include <GUI/BsCGUIWidget.h>
+#include <GUI/BsGUILabel.h>
+#include <GUI/BsGUIPanel.h>
+#include <Math/BsMath.h>
+#include <Scene/BsSceneObject.h>
+#include <excepction/Throw.hpp>
+
+namespace REGoth
+{
+  AnchoredTextLabels::AnchoredTextLabels(const bs::HSceneObject& parent)
+      : Component(parent)
+  {
+    // Set a name for the component, so we can find it later if needed
+    setName("AnchoredTextLabels");
+
+    // A GUI widget component is required for drawing the labels
+    mGui = parent->getComponent<bs::CGUIWidget>();
+
+    if (mGui.isDestroyed())
+    {
+      REGOTH_THROW(
+          InvalidStateException,
+          bs::StringUtil::format("Scene Object {0} does not have a CGUIWidget component!",
+                                 SO()->getName()));
+    }
+  }
+
+  void AnchoredTextLabels::addLabel(const bs::Vector3& anchorPosition, const bs::HString& text)
+  {
+    bs::GUILabel* label = bs::GUILabel::create(text);
+    mLabels.emplace_back(AnchoredLabel{anchorPosition, label});
+
+    bs::GUIPanel* mainPanel = mGui->getPanel();
+    mainPanel->addElement(label);
```

Comment:
Should also remove those elements again in the destructor.

Reply:
Good catch. When I want to remove the labels in the destructor, I have to get the `mainPanel` again from `mGui`. Is there a guarantee that the `CGUIWidget` component is destroyed *after* mine if it has been added to the scene object before mine (e. g. in the case of the application being shut down)? If there is not, I should probably check if `mGui` is already destroyed before trying to remove the labels.

Reply:
An alternative would be using the `onDestroyed()` method. Would probably be the best alternative as the bs::Component documentation says 
> Avoid putting logic in constructors or destructors.

Reply:
`onDestroyed` seems to be the correct way to go. If the `CGUI` is destroyed, you won't need to remove any labels anyways. You should still be checking whether the handle to `mGui` is valid to not cause an exception when accessing the destroyed object.

Reply:
Fixed in 3dbd4fd.